### PR TITLE
Adds method to create unique-per-asset links based on title

### PR DESCRIPTION
**Pull request by cmgrote**

### CODE REVIEW

File: pyatlan/generator/templates/methods/asset/asset.jinja2 [hunk 1]
```diff
@@ -7,6 +7,21 @@
...
+        if bool(
+            self.unique_attributes
+            and self.unique_attributes.get("qualified_name")
+            and str(self.unique_attributes.get("qualified_name")).strip()
+        ):
+            return self.ref_by_qualified_name(
+                str(cast(dict[str, Any], self.unique_attributes).get("qualified_name"))
+            )
+        raise ErrorCode.MISSING_REQUIRED_RELATIONSHIP_PARAM.exception_with_parameters(self.type_name, "guid, qualifiedName")
+
```
Comment (ErnestoLoma) [inline pyatlan/generator/templates/methods/asset/asset.jinja2:24]: The bools are superfluous and I believe you are using str to get around the mypy warning that the qualified_name is  Optional. I think the following is cleaner:
```
        if self.guid and self.guid.strip():
            return self.ref_by_guid(self.guid)
        if self.qualified_name and self.qualified_name.strip():
            return self.ref_by_qualified_name(self.qualified_name)
        if (
            self.unique_attributes
            and (qualified_name := self.unique_attributes.get("qualified_name"))
            and qualified_name.strip()
        ):
            return self.ref_by_qualified_name(qualified_name)
        raise ErrorCode.MISSING_REQUIRED_RELATIONSHIP_PARAM.exception_with_parameters(
            self.type_name, "guid, qualifiedName"
        )
```

**Review by ErnestoLoma — CHANGES_REQUESTED** [review]
One change

File: pyatlan/generator/templates/methods/asset/asset.jinja2 (hunk 1 above)
Comment (cmgrote) [reply pyatlan/generator/templates/methods/asset/asset.jinja2]: Yes, thanks! Seemed a bit ridiculous before 😬

**Review by ErnestoLoma — APPROVED** [review]
Looks good

**Pull request merged**